tangelo/toolboxes/ansatz_generator/tests/test_adapt_ansatz.py

```diff
@@ -57,6 +58,37 @@ def test_adaptansatz_set_var_params(self):
         with self.assertRaises(ValueError):
             ansatz.set_var_params([1.999, 2.999])
 
+    def test_adaptansatz_reference_state_circuit(self):
+        """ Verify variational gate parameters are frozen in a reference state circuit"""
+
+        # Create simple reference circuit:
+        ref_circuit = Circuit([
+            Gate('RY', 0, parameter=1.0, is_variational=True)
+            for _ in range(4)
```

Comment:
I don't know if the circuit you wanted to prepend was RY rotations on 4 different qubits, but here it is 4 RY on the first qubit (so with=1). If we change the reference circuit to

```python
 ref_circuit = Circuit([
            Gate('RY', i, parameter=1.0, is_variational=True)
            for i in range(4)
    ])
```

We could move `self.assertEqual(adapt_circ.width, 4)` after `build_circuit`, which would ensure the circuit has been prepend. Now, we are relying on `qu_op` to test the width.

* We can also keep the same reference circuit and test `self.assertEqual(adapt_circ.width, 1)`, that could be another option.

Reply:
My mistake, thanks for catching that! I have fixed this in [commit dc09e9f](https://github.com/goodchemistryco/Tangelo/pull/398/commits/dc09e9f19c035bab984c783fd81dbba6ae828767). Can you take a look and make sure it is correct now?

Reply:
All good!